packages/SwingSet/src/kernel/kernel.js

```diff
@@ -263,6 +265,18 @@ export default function buildKernel(
       // eslint-disable-next-line no-use-before-define
       vatWarehouse.vatWasTerminated(vatID);
     }
+    if (critical) {
+      // The following error construction is a bit awkward, but (1) it saves us
+      // from doing unmarshaling while in the kernel, while (2) it protects
+      // against the info not actually encoding an error, but (3) it still
+      // provides some diagnostic information back to the host, and (4) this
+      // should happen rarely enough that if you have to do a little bit of
+      // extra interpretive work on the receiving end to diagnose the problem,
+      // it's going to be a small cost compared to the trouble you're probably
+      // already in anyway if this happens.
+      panic(`critical vat failure ${vatID}`, Error(info.body));
```

Comment:
I might reword this slightly, to `critical vat ${vatID} failed`. The original could be read to mean "a vat failed in a particularly critical way", as opposed to "a vat, which was marked critical, has suffered an (ordinary) failure".